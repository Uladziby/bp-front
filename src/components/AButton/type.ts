/** @format */

import { MouseEvent, ReactNode } from "react";

export interface IAButtonProps {
	onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
	variant?: "primary" | "secondary";
	isDisabled?: boolean;
	size?: "extraSmall" | "small" | "medium" | "large";
	children: string | ReactNode;
	type?: "button" | "submit" | "reset";
	link?: string;
	className?: string;
	isActive?: boolean;
	iconTitle?: string;
	dataTestId?: string;
	isWithMotion?: boolean;
	ishidden?: boolean;
}
