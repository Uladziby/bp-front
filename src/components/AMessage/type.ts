import { ReactNode } from 'react';

export interface IAMessageProps {
  children: string | ReactNode;
  type: 'warning' | 'error';
  className?: string;
}
