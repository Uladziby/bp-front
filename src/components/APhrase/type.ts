/** @format */

export interface IAPhraseProps {
	color?: string;
	size: number;
	weight: number;
	lineHeight: number;
	children: string;
	className?: string;
	fontFamily: "Cormorant Garamond" | "Manrope";
	dataTestId?: string;
}
