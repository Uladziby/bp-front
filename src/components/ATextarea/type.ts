export interface IATextarea {
  placeholder: string;
  label: string;
  hasError?: boolean;
  isDisabled?: boolean;
  name: string;
  defaultValue?: string;
  maxLength?: number;
  rows?: number;
}
