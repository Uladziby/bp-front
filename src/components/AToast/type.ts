/** @format */

import { ReactNode } from "react";

export interface AToastProps {
	title?: string;
	isOpen: boolean;
	context?: ReactNode | string;
	icon?: ReactNode;
}
