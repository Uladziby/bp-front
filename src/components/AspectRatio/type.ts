/** @format */

export interface AspectRatioProps {
	image: string;
	ratio: number;
	width: number;
	height?: number;
}
