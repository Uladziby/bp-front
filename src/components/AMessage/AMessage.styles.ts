/** @format */

import styled from "styled-components";
import { COLOR_PALETTE } from "utils/colors";

const { error, gray150, gray900 } = COLOR_PALETTE;

export const StyledContainer = styled.strong<{ type: string }>`
	display: block;
	text-align: left;
	font-weight: 400;
	font-size: 12px;
	line-height: 16px;
	color: ${({ type }) => (type === "error" ? error : gray900)};
	background-color: ${({ type }) => (type === "error" ? "transparent" : gray150)};
	padding: ${({ type }) => (type === "error" ? "0" : "8px 12px")};
`;
