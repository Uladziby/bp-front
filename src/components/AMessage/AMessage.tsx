import { StyledContainer } from './AMessage.styles';
import { IAMessageProps } from './type';

export const AMessage = ({ children, type, className }: IAMessageProps) => (
  <StyledContainer data-testid="aMessage" className={className} type={type}>
    {children}
  </StyledContainer>
);
