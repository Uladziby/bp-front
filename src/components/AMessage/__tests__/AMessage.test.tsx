/** @format */

import { RenderResult, render, screen } from "@testing-library/react";
import { AMessage } from "../AMessage";
import { IAMessageProps } from "../type";
import { COLOR_PALETTE } from "utils/colors";

describe("AMessage", () => {
	const props: IAMessageProps = {
		children: "child",
		type: "warning",
	};

	let renderResult: RenderResult;

	beforeEach(() => {
		renderResult = render(<AMessage {...props} />);
	});

	it("should render correctly", () => {
		const element = screen.getByTestId("aMessage");

		expect(element).toBeInTheDocument();
		expect(element).toHaveTextContent("child");
	});

	it("snapshot", () => {
		const element = screen.getByTestId("aMessage");

		expect(element).toMatchSnapshot();
	});

	it("should render correctly with Node child", () => {
		renderResult.rerender(
			<AMessage type="warning">
				<a href="#">Test</a>
			</AMessage>
		);

		const element = screen.getByTestId("aMessage");

		expect(element).toBeInTheDocument();
		expect(element.innerHTML).toEqual('<a href="#">Test</a>');
	});

	it("should render correctly with multiple children", () => {
		renderResult.rerender(
			<AMessage type="warning">
				test <a href="#">Test</a> test <a href="#">Test</a>
			</AMessage>
		);

		const element = screen.getByTestId("aMessage");

		expect(element).toBeInTheDocument();
		expect(element.innerHTML).toEqual('test <a href="#">Test</a> test <a href="#">Test</a>');
	});

	it('should apply correct styling when prop "type" = "warning"', () => {
		const element = screen.getByTestId("aMessage");

		expect(element).toHaveStyle(`color: ${COLOR_PALETTE.gray900}`);
		expect(element).toHaveStyle(`background-color: ${COLOR_PALETTE.gray150}`);
		expect(element).toHaveStyle("padding: 8px 12px");
	});

	it('should apply correct styling when prop "type" = "error"', () => {
		renderResult.rerender(<AMessage type="error">text</AMessage>);

		const element = screen.getByTestId("aMessage");

		expect(element).toHaveStyle(`color: ${COLOR_PALETTE.error}`);
		expect(element).toHaveStyle("background-color: transparent");
		expect(element).toHaveStyle("padding: 0");
	});
});
