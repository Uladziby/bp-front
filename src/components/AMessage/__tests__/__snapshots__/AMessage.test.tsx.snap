// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMessage snapshot 1`] = `
<strong
  class="sc-bcXHqe iIfYyV"
  data-testid="aMessage"
  type="warning"
>
  child
</strong>
`;
